src/test/antoraSupport.test.ts

```diff
@@ -0,0 +1,60 @@
+import * as assert from 'assert'
+import 'mocha'
+import * as vscode from 'vscode'
+import { getAntoraConfig } from '../features/antora/antoraSupport'
+
+let root
+
+suite('Antora Support', () => {
+  setup(() => {
+    root = vscode.workspace.workspaceFolders[0].uri.fsPath
+  })
+  test('Should return Antora config', async () => {
+    const expectedPath = `${root}/antora/multiComponents/api/antora.yml`
+    const file = await vscode.workspace.openTextDocument(vscode.Uri.file(`${root}/antora/multiComponents/api/modules/auth/pages/jwt/page2.adoc`))
+    const actualUri = await getAntoraConfig(file)
+    assert.strictEqual(actualUri.fsPath, expectedPath)
+  })
+
+  test('Should return Antora config 2', async () => {
+    const expectedPath = `${root}/antora/multiComponents/api/antora.yml`
+    const file = await vscode.workspace.openTextDocument(vscode.Uri.file(`${root}/antora/multiComponents/api/modules/auth/pages/page3.adoc`))
+    const actualUri = await getAntoraConfig(file)
+    assert.strictEqual(actualUri.fsPath, expectedPath)
+  })
+
+  test('Should return Antora config for document inside a "modules" subfolder', async () => {
+    const expectedPath = `${root}/antora/multiComponents/api/antora.yml`
+    const file = await vscode.workspace.openTextDocument(vscode.Uri.file(`${root}/antora/multiComponents/api/modules/auth/pages/modules/page4.adoc`))
+    const actualUri = await getAntoraConfig(file)
+    assert.strictEqual(actualUri.fsPath, expectedPath)
+  })
+
+  test('Should return Antora config for document inside a "modules" directory, inside an Antora modules in a composant named "modules"', async () => {
+    const expectedPath = `${root}/antora/multiComponents/modules/api/docs/antora.yml`
+    const file = await vscode.workspace.openTextDocument(vscode.Uri.file(`${root}/antora/multiComponents/modules/api/docs/modules/asciidoc/pages/modules/page5.adoc`))
+    const actualUri = await getAntoraConfig(file)
+    assert.strictEqual(actualUri.fsPath, expectedPath)
+  })
+
+  test('Should return Antora config for document inside a directory wich has the same name as the workspace', async () => {
+    const expectedPath = `${root}/antora/multiComponents/api/antora.yml`
+    const file = await vscode.workspace.openTextDocument(vscode.Uri.file(`${root}/antora/multiComponents/api/modules/auth/pages/modules/multiComponents/page6.adoc`))
+    const actualUri = await getAntoraConfig(file)
+    assert.strictEqual(actualUri.fsPath, expectedPath)
+  })
+
+  test('Should not return Antora config for document outside "modules" Antora folder', async () => {
+    const expected = undefined
+    const file = await vscode.workspace.openTextDocument(vscode.Uri.file(`${root}/antora/multiComponents/api/modules/writer-guide.adoc`))
+    const actualUri = await getAntoraConfig(file)
+    assert.strictEqual(actualUri, expected)
+  })
+
+  test('Should not return Antora config for document outside of workspace', async () => {
```

Comment:
since we are doing the same thing we could introduce a function:

```ts
async function testGetAntoraConfig (asciidocPath, antoraConfigExpectedFsPath) {
    const file = await vscode.workspace.openTextDocument(vscode.Uri.file(`${root}/${asciidocPath}`))
    const antoraConfigUri = await getAntoraConfig(file)
    if (antoraConfigExpectedPath === undefined) {
        assert.strictEqual(antoraConfigUri, undefined)
    } else {
        assert.strictEqual(antoraConfigUri.fsPath, antoraConfigExpectedFsPath)
    }
}
```

Then we could call it in each test:

```ts
test('Should return Antora config', async () => testGetAntoraConfig(
  'antora/multiComponents/api/modules/auth/pages/jwt/page2.adoc',
  `${root}/antora/multiComponents/api/antora.yml`
))

test('Should return Antora config for document inside a directory wich has the same name as the workspace', async () => testGetAntoraConfig(
  'antora/multiComponents/api/modules/auth/pages/modules/multiComponents/page6.adoc',
  `${root}/antora/multiComponents/api/antora.yml`
))

test('Should not return Antora config for document outside "modules" Antora folder', async () => testGetAntoraConfig(
  'antora/multiComponents/api/modules/writer-guide.adoc',
  undefined
))
```

We can even generate the `test` block by extracting test cases data:

```ts
const testCases = [
  {
    title: 'Should return Antora config',
    asciidocPath: 'antora/multiComponents/api/modules/auth/pages/jwt/page2.adoc',
    antoraConfigExpectedFsPath: `${root}/antora/multiComponents/api/antora.yml`
  }
  // ...
]
for (const testCase of testCases) {
  test(testCase.title, async () => testGetAntoraConfig(testCase.asciidocPath, testCase.antoraConfigExpectedFsPath))
}
```